services/auth_service.py

```
@@ -16,29 +18,47 @@ def verify_password(plain_password : str, hashed_password :str):
 def signup_user(data: UserSignUp):
-    users = get_users()
-
-    if any(u["email"] == data.email for u in users):
-        raise HTTPException(400, "이미 존재하는 이메일")
+    con = get_db_connection()
+    try:
+        with con.cursor() as cursor:
+            check_sql = "SELECT * FROM users WHERE email = %s"
+            cursor.execute(check_sql, (data.email, ))
+            if cursor.fetchone():
+                raise HTTPException(status_code=400, detail="이미 존재하는 이메일")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)